packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -388,4 +392,25 @@ export class BaseInvocationScope<TReturn = any> {
     const transactionRequest = await this.getTransactionRequest();
     return transactionRequest.getTransactionId(chainIdToHash);
   }
+
+  /**
+   * In case the gasLimit and gasPrice are *not* set by the user, this method sets some default values.
+   */
+  async setDefaultTxParams() {
+    const transactionRequest = await this.getTransactionRequest();
+    const { minGasPrice } = this.getProvider().getGasConfig();
+    const { gasUsed } = await this.getTransactionCost();
+
+    const specifiedGasLimit = this.txParameters?.gasLimit || this.hasCallParamsGasLimit;
+    const specifiedGasPrice = this.txParameters?.gasPrice;
+
+    // set defaults for gasLimit and gasPrice if not specified
+    if (!specifiedGasLimit) {
+      transactionRequest.gasLimit = gasUsed;
+    }
+
+    if (!specifiedGasPrice) {
+      transactionRequest.gasPrice = minGasPrice;
+    }
```

Comment:
```suggestion
    const transactionRequest = await this.getTransactionRequest();

    const gasLimitSpecified = this.txParameters?.gasLimit || this.hasCallParamsGasLimit;
    const gasPriceSpecified = this.txParameters?.gasPrice;

    // set defaults for gasLimit and gasPrice if not specified
    if (!gasLimitSpecified) {
      const { gasUsed } = await this.getTransactionCost();
      transactionRequest.gasLimit = gasUsed;
    }

    if (!gasPriceSpecified) {
      const { minGasPrice } = this.getProvider().getGasConfig();
      transactionRequest.gasPrice = minGasPrice;
    }
```
Two things:
- Rename `specifiedGasLimit`/`specifiedGasPrice` to `gasLimitSpecified`/`gasPriceSpecified`. The initial naming sounds like it's holding a numerical value while the suggested naming implies a boolean.
- Move `gasUsed` and `minGasPrice` into the `if` blocks where they're used.